infinispan 16.0.9 server with rocksdb does not works in docker mode.

### Describe the bug

I was previously using infinispan 15.2.6 with rocksdb, it was working fine. But after I switched to 16.0.9 I started getting error when creating a cache with persistance


```
<distributed-cache statistics="true" owners="1">\    
<encoding media-type="application/x-protostream"/>\    
<locking striping="false" concurrency-level="256"/>\    
<persistence>\        
<rocksdb-store xmlns="urn:infinispan:config:store:rocksdb:16.0" path="publish-status-cache-rocksdb" shared="false" purge="false" transactional="false">\            
<compression type="SNAPPY"/>\        
</rocksdb-store>\    
</persistence>\    
<state-transfer await-initial-transfer="false" />\  
</distributed-cache>
```
Error I get is

```
Caused by: java.lang.NoClassDefFoundError: org/rocksdb/CompressionType
infinispan-1  |     at org.infinispan.persistence.rocksdb.configuration.CompressionType.<clinit>(CompressionType.java:4)
infinispan-1  |     at org.infinispan.persistence.rocksdb.configuration.RocksDBStoreConfiguration.<clinit>(RocksDBStoreConfiguration.java:23)
infinispan-1  |     at org.infinispan.persistence.rocksdb.configuration.RocksDBStoreConfigurationBuilder.<init>(RocksDBStoreConfigurationBuilder.java:23)
infinispan-1  |     at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
infinispan-1  |     ... 28 more
infinispan-1  | Caused by: java.lang.ClassNotFoundException: org.rocksdb.CompressionType
infinispan-1  |     at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
infinispan-1  |     at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
infinispan-1  |     ... 32 more
```
Below is my docker file

```
FROM quay.io/infinispan/server:16.0.9

RUN echo '<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0">' \
         '<profiles>' \
           '<profile>' \
             '<id>infinispan</id>' \
             '<repositories>' \
               '<repository>' \
                 '<id>infinispan</id>' \
                 '<url>https://repository.infinispan.org/</url>' \
               '</repository>' \
               '<repository>' \
                 '<id>central</id>' \
                 '<url>https://repo1.maven.org/maven2/</url>' \
               '</repository>' \
             '</repositories>' \
           '</profile>' \
         '</profiles>' \
         '<activeProfiles>' \
           '<activeProfile>infinispan</activeProfile>' \
         '</activeProfiles>' \
       '</settings>' > /tmp/settings.xml

ENV MAVEN_OPTS="-s /tmp/settings.xml"

RUN /opt/infinispan/bin/cli.sh install org.rocksdb:rocksdbjni:9.0.1
RUN /opt/infinispan/bin/cli.sh install org.infinispan:infinispan-cachestore-rocksdb:16.0.9

RUN cp /opt/infinispan/server/lib/rocksdbjni-9.0.1.jar /opt/infinispan/lib/
RUN cp /opt/infinispan/server/lib/infinispan-cachestore-rocksdb-16.0.9.jar /opt/infinispan/lib/

USER root
RUN chown -R 1000:1000 /opt/infinispan/server
USER 1000:1000

ENTRYPOINT ["/opt/infinispan/bin/launch.sh"]
```
I have confirmed that rocksdbjni jar is present in /opt/infinispan/lib and /opt/infinispan/server/lib

### Expected behavior

cache creation with rocksdb persistance should work

### Actual behavior

cache creation fails 

### How to Reproduce?

1. Use the docker file pasted above to create a docker image
2. run docker image to start server
3. Use any client to try to create cache based on cache config above

--- a/documentation/pom.xml
+++ b/documentation/pom.xml
@@ -261,6 +261,7 @@
                                 <code_tutorials>${infinispan.tutorials}</code_tutorials>
                                 <imagesdir>../../topics/images</imagesdir>
                                 <postgresqlversion>${postgresql.database.version}</postgresqlversion>
+                                <rocksdbversion>${version.rocksdb}</rocksdbversion>
                                 <source-highlighter>highlight.js</source-highlighter>
                                 <highlightjs-theme>github</highlightjs-theme>
                             </attributes>
@@ -334,6 +335,7 @@
                                         <code_tutorials>${infinispan.tutorials}</code_tutorials>
                                         <imagesdir>../../topics/images</imagesdir>
                                         <postgresqlversion>${postgresql.database.version}</postgresqlversion>
+                                        <rocksdbversion>${version.rocksdb}</rocksdbversion>
                                         <source-highlighter>pygments</source-highlighter>
                                     </attributes>
                                 </configuration>

--- a/documentation/src/main/asciidoc/topics/changes.adoc
+++ b/documentation/src/main/asciidoc/topics/changes.adoc
@@ -470,11 +470,13 @@ NOTE: This change has no effect for shared stores
 
 === RocksDB store
 
-The RocksDB implementation is no longer included by default with the server. It can be installed via the CLI:
+The RocksDB cache store is no longer included by default with the server.
+You must install both the {brandname} RocksDB module and the RocksDB native library using the CLI:
 
 [source,shell,options="nowrap",subs=attributes+]
 ----
-cli.sh install org.rocksdb:rocksdbjni:6.22.1.1
+[disconnected]> install org.infinispan:infinispan-cachestore-rocksdb:{ispn_version}
+[disconnected]> install org.rocksdb:rocksdbjni:{rocksdbversion}
 ----
 
 === JDBC Store database versions configuration deprecated

--- a/documentation/src/main/asciidoc/topics/ref_cache_store_rocksdb.adoc
+++ b/documentation/src/main/asciidoc/topics/ref_cache_store_rocksdb.adoc
@@ -7,6 +7,27 @@ RocksDB cache stores, `RocksDBStore`, use two databases. One database provides
 a primary cache store for data in memory; the other database holds entries that
 {brandname} expires from memory.
 
+.Server installation
+
+The RocksDB cache store is not included in the {brandname} Server distribution by default.
+You must install both the {brandname} RocksDB cache store module and the RocksDB native library before you can configure RocksDB cache stores.
+
+Use the {brandname} CLI to install the required artifacts to the `server/lib` directory:
+
+[source,shell,options="nowrap",subs=attributes+]
+----
+[disconnected]> install org.infinispan:infinispan-cachestore-rocksdb:{ispn_version}
+[disconnected]> install org.rocksdb:rocksdbjni:{rocksdbversion}
+----
+
+You must restart the server after installing the artifacts.
+
+[NOTE]
+====
+The RocksDB native library version shown above is the version that was tested with this release of {brandname}.
+Newer versions of the library may or may not be compatible.
+====
+
 .Configuration parameters
 |===
 | Parameter | Description

--- a/server/core/src/main/java/org/infinispan/server/core/ServerManagement.java
+++ b/server/core/src/main/java/org/infinispan/server/core/ServerManagement.java
@@ -72,4 +72,5 @@ public interface ServerManagement {
 
    Json securityOverviewReport();
 
+    ClassLoader getClassLoader();
 }

--- a/server/core/src/test/java/org/infinispan/server/core/DummyServerManagement.java
+++ b/server/core/src/test/java/org/infinispan/server/core/DummyServerManagement.java
@@ -128,4 +128,9 @@ public class DummyServerManagement extends BaseServerManagement {
    public Json securityOverviewReport() {
       return Json.object();
    }
+
+   @Override
+   public ClassLoader getClassLoader() {
+      return Thread.currentThread().getContextClassLoader();
+   }
 }

--- a/server/rest/src/main/java/org/infinispan/rest/InvocationHelper.java
+++ b/server/rest/src/main/java/org/infinispan/rest/InvocationHelper.java
@@ -25,7 +25,7 @@ import io.netty.handler.codec.http.HttpResponseStatus;
  * @since 10.0
  */
 public class InvocationHelper {
-   private final ParserRegistry parserRegistry = new ParserRegistry();
+   private final ParserRegistry parserRegistry;
    private final RestCacheManager<Object> restCacheManager;
    private final RestServerConfiguration configuration;
    private final ServerManagement server;
@@ -38,6 +38,7 @@ public class InvocationHelper {
 
    public InvocationHelper(RestServer protocolServer, RestCacheManager<Object> restCacheManager,
                     RestServerConfiguration configuration, ServerManagement server, Executor executor) {
+      this.parserRegistry = new ParserRegistry(server.getClassLoader());
       this.protocolServer = protocolServer;
       this.restCacheManager = restCacheManager;
       this.configuration = configuration;

--- a/server/rest/src/main/java/org/infinispan/rest/resources/OpenAPIResource.java
+++ b/server/rest/src/main/java/org/infinispan/rest/resources/OpenAPIResource.java
@@ -247,6 +247,11 @@ public class OpenAPIResource implements ResourceHandler {
             public Json securityOverviewReport() {
                return null;
             }
+
+            @Override
+            public ClassLoader getClassLoader() {
+               return Thread.currentThread().getContextClassLoader();
+            }
          }, null);
          ResourceManager resourceManager = new ResourceManagerImpl();
          OpenAPIResource openAPIResource = new OpenAPIResource(invocationHelper, resourceManager.registry());

--- a/server/runtime/pom.xml
+++ b/server/runtime/pom.xml
@@ -96,16 +96,6 @@
       </dependency>
       <dependency>
          <groupId>org.infinispan</groupId>
-         <artifactId>infinispan-cachestore-rocksdb</artifactId>
-         <exclusions>
-            <exclusion>
-               <groupId>org.rocksdb</groupId>
-               <artifactId>rocksdbjni</artifactId>
-            </exclusion>
-         </exclusions>
-      </dependency>
-      <dependency>
-         <groupId>org.infinispan</groupId>
          <artifactId>infinispan-cli-client</artifactId>
       </dependency>
       <dependency>

--- a/server/runtime/src/main/java/org/infinispan/server/Bootstrap.java
+++ b/server/runtime/src/main/java/org/infinispan/server/Bootstrap.java
@@ -224,7 +224,7 @@ public class Bootstrap extends Main {
    }
 
    private void configureClasspath() {
-      ClassLoader serverClassLoader = null;
+      ClassLoader serverClassLoader = Thread.currentThread().getContextClassLoader();
       String lib = properties.getProperty(Loader.INFINISPAN_SERVER_LIB_PATH);
       if (lib != null) {
          for (String item : lib.split(File.pathSeparator)) {

--- a/server/runtime/src/main/java/org/infinispan/server/Server.java
+++ b/server/runtime/src/main/java/org/infinispan/server/Server.java
@@ -802,6 +802,11 @@ public class Server extends BaseServerManagement implements AutoCloseable {
    }
 
    @Override
+   public ClassLoader getClassLoader() {
+      return classLoader;
+   }
+
+   @Override
    public CompletionStage<Void> flushSecurityCaches() {
       return SecurityActions.getClusterExecutor(cacheManager)
             .submitConsumer(ecm -> {

--- a/server/tests/pom.xml
+++ b/server/tests/pom.xml
@@ -252,6 +252,7 @@
                   <infinispan.module-suffix>${infinispan.module-suffix}</infinispan.module-suffix>
                   <ansi.strip>${ansi.strip}</ansi.strip>
                   <version.bouncycastle>${version.bouncycastle}</version.bouncycastle>
+                  <version.rocksdb>${version.rocksdb}</version.rocksdb>
                </systemPropertyVariables>
                <trimStackTrace>false</trimStackTrace>
                <testNGArtifactName>none:none</testNGArtifactName>

--- a/server/tests/src/test/java/org/infinispan/server/persistence/RocksDBStoreIT.java
+++ b/server/tests/src/test/java/org/infinispan/server/persistence/RocksDBStoreIT.java
@@ -1,0 +1,54 @@
+package org.infinispan.server.persistence;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.commons.configuration.StringConfiguration;
+import org.infinispan.commons.util.Version;
+import org.infinispan.server.test.core.ServerRunMode;
+import org.infinispan.server.test.core.tags.Persistence;
+import org.infinispan.server.test.junit5.InfinispanServerExtension;
+import org.infinispan.server.test.junit5.InfinispanServerExtensionBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@Persistence
+public class RocksDBStoreIT {
+
+   @RegisterExtension
+   public static InfinispanServerExtension SERVERS =
+         InfinispanServerExtensionBuilder.config("configuration/CustomStoreTest.xml")
+               .numServers(1)
+               .runMode(ServerRunMode.CONTAINER)
+               .mavenArtifacts("org.infinispan:infinispan-cachestore-rocksdb:" + Version.getVersion(),
+                     "org.rocksdb:rocksdbjni:" + System.getProperty("version.rocksdb"))
+               .build();
+
+   @Test
+   public void testRocksDBStore() {
+      String cacheName = "rocksdb-cache";
+      String config = String.format("""
+            <distributed-cache name="%s">
+               <persistence>
+                  <rocksdb-store xmlns="urn:infinispan:config:store:rocksdb:%s"/>
+               </persistence>
+            </distributed-cache>""", cacheName, Version.getMajorMinor());
+      RemoteCache<String, String> cache = SERVERS.hotrod().withServerConfiguration(new StringConfiguration(config)).create();
+
+      cache.put("k1", "v1");
+      cache.put("k2", "v2");
+      cache.put("k3", "v3");
+      assertEquals("v1", cache.get("k1"));
+      assertEquals("v2", cache.get("k2"));
+      assertEquals("v3", cache.get("k3"));
+      assertEquals(3, cache.size());
+
+      cache.put("k2", "v2-updated");
+      assertEquals("v2-updated", cache.get("k2"));
+
+      cache.remove("k1");
+      assertNull(cache.get("k1"));
+      assertEquals(2, cache.size());
+   }
+}

